### PR TITLE
Bump go-containerregistry to latest version

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This allows to fix the OpenShit + GCP bug (#1742) **and** it also removes the
dependency on `k8s.io/kubernetes` that is a pain when upgrading to new
kubernetes version.

See https://github.com/google/go-containerregistry/pull/649

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](http #1742s://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,31 +32,25 @@ require (
...
 	google.golang.org/api v0.10.0 // indirect
 	google.golang.org/appengine v1.6.5 // indirect
-	google.golang.org/grpc v1.24.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.2.5 // indirect
-	k8s.io/api v0.0.0
-	k8s.io/apimachinery v0.0.0
-	k8s.io/client-go v0.0.0
-	k8s.io/code-generator v0.0.0
+	k8s.io/api v0.17.0
```
Comment (vdemeester) [inline go.mod]: Can't make `go` not changing thoses (`go mod tidy` & co always add this one…) It doesn't matter as they are replaced.. it's just a pain…

**Comment by vdemeester** [comment]
/hold

Something is fishy though.. I don't get why there is `docker/docker`, `containerd/containerd`, … in the vendor folder (and not in the `go.mod`...)…

**Comment by vdemeester** [comment]
oups.. I need to add a `LICENSE` to my repository…

**Comment by vdemeester** [comment]
```
# github.com/docker/docker/api/types
github.com/tektoncd/pipeline/pkg/pod
github.com/google/go-containerregistry/pkg/authn/k8schain
github.com/vdemeester/k8s-pkg-credentialprovider
github.com/docker/docker/api/types
```

Interesting.. which version did it take…

File: go.mod (hunk 1 above)
Comment (dibyom) [reply go.mod]: So, we have to keep replacing them manually?

Comment (vdemeester) [reply go.mod]: For now I think so… I am hoping we won't, I need to try something out.. but I think until we bump to 1.15 (v0.15) or 1.17, we will still need the replace (but I might be wrong, I'll try that when I bump to 1.15 :angel: )

**Pull request merged**